### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by SerjDSA**
[DEMO LINK](https://serjdsa.github.io/react_add-todo-form/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,61 +1,44 @@
...
+  email: string;
+};
+
+type TodoItem = {
+  id: number;
+  title: string;
+  completed: boolean;
+  userId: number;
+  user?: User;
+};
```
Comment (etojeDenys) [inline src/App.tsx:20]: move it to some type file and import here

File: src/components/TodoInfo/TodoInfo.tsx [hunk 2]
```diff
@@ -1 +1,38 @@
...
+  id: number;
+  title: string;
+  completed: boolean;
+  userId: number;
+  user?: User;
+};
+
+interface TodoInfoProps {
+  todo: TodoItem;
+}
```
Comment (etojeDenys) [inline src/components/TodoInfo/TodoInfo.tsx:21]: same here, just import the types

File: src/components/UserInfo/UserInfo.tsx [hunk 3]
```diff
@@ -1 +1,22 @@
...
+export const UserInfo: React.FC<UserInfoProps> = ({ user }) => {
+  return (
+    <>
+      {user && (
+        <a className="UserInfo" href={`mailto:${user.email}`}>
+          {user.name}
+        </a>
+      )}
+    </>
+  );
```
Comment (etojeDenys) [inline src/components/UserInfo/UserInfo.tsx:21]:
```suggestion
interface UserInfoProps {
  user: UsersItem;
}

export const UserInfo: React.FC<UserInfoProps> = ({ user }) => {
  return (
        <a className="UserInfo" href={`mailto:${user.email}`}>
          {user.name}
        </a>
  );
```
let's remove the check here, and render only when user exists

File: src/components/TodoInfo/TodoInfo.tsx [hunk 4]
```diff
@@ -1 +1,38 @@
...
+  return (
+    <article
+      key={id}
+      data-id={id}
+      className={classNames('TodoInfo', {
+        'TodoInfo--completed': completed,
+      })}
+    >
+      <h2 className="TodoInfo__title">{title}</h2>
+      {todo && <UserInfo user={user} />}
```
Comment (etojeDenys) [inline src/components/TodoInfo/TodoInfo.tsx:35]:
```suggestion
      {user && <UserInfo user={user} />}
```

**Review by etojeDenys — CHANGES_REQUESTED** [review]

**Review by SerjDSA — COMMENTED** [review]
[DEMO LINK](https://serjdsa.github.io/react_add-todo-form/)

**Pull request closed without merging**